### PR TITLE
Push Notification Proof of Concept

### PR DESCRIPTION
**Pull request by JacobJaffe**
Some general fixes in first few commits -- see ` js/screens/settings/SettingsNotificationsScreen.js ` for the main stuff here.

Proof of concept of the different case handling for notifications. 

This all occurs in a component right now, but handles the cases exhaustively, mocking an API request that fails every other time. 

Go to Settings -> Notifications in app and play around with the toggles to see how the cases are handled. 

Basically, on api failure the settings revert back. 

Note that this _doesn't_ account for triggers by language -- changing a language doesn't fire off the mock api. Also, the state isn't persisted -- this should be managed via redux eventually.

### CODE REVIEW

**Comment by JacobJaffe** [comment]
Bumped expo to 40 in this also now, to ensure compatibility testing with Expo Client

File: js/components/ListSelector.js [hunk 1]
```diff
@@ -64,6 +65,7 @@ const ListSelector = ({
   return (
     <FlatList
       alwaysBounceVertical={false}
+      contentContainerStyle={contentContainerStyle}
```
Comment (kiwansim) [inline js/components/ListSelector.js:68]: bless this modularity

**Review by kiwansim — APPROVED** [review]

**Review by kiwansim — COMMENTED** [review]
@JacobJaffe one change here!
